io/src/ply_io.cpp

```diff
@@ -358,7 +358,7 @@ namespace pcl
         cloud_->point_step = static_cast<std::uint32_t> (std::numeric_limits<std::uint32_t>::max ());
       do_resize_ = true;
       return std::tuple<std::function<void (SizeType)>, std::function<void (ContentType)>, std::function<void ()> > (
-        std::bind (&pcl::PLYReader::vertexListPropertyBeginCallback<SizeType>, this, property_name, std::placeholders::_1),
+        [this, property_name](auto && PH1) { this->vertexListPropertyBeginCallback(property_name, std::forward<decltype(PH1)>(PH1)); },
```

Comment:
```suggestion
        [this, property_name](SizeType size) { vertexListPropertyBeginCallback(property_name, size); },
```
Otherwise looks good.

Reply:
I added the `this->` because, on the macOS compile, it complains that the `this` in the capture is not used. I added a comment to that effect.

Reply:
That is strange. I assume you are using AppleClang? Which version? Could that be a compiler bug? I tried to replicate the warning on godbolt (https://godbolt.org/z/qdWdKdW9j), but couldn't.
What about the other lambdas, e.g. lines 342-344, does it complain about those, too? Because they also capture `this` but do not explicitly use `this->` in the lambda body

Reply:
> I assume you are using AppleClang?

The failure was in CI, but I can't find it in the log. I'll try pushing the change you suggest and see if it happens again.